src/init.lua

```diff
@@ -8,13 +8,24 @@ local makeThunkMiddleware = require(script.makeThunkMiddleware)
 
 local actions = require(script.types.actions)
 local reducers = require(script.types.reducers)
+local store = require(script.types.store)
+local thunks = require(script.types.thunks)
 
 export type Action<Type = any> = actions.Action<Type>
 export type AnyAction = actions.AnyAction
 export type ActionCreator<Type, Action, Args...> = actions.ActionCreator<Type, Action, Args...>
 
 export type Reducer<State = any, Action = AnyAction> = reducers.Reducer<State, Action>
 
+export type BoundDispatch = store.BoundDispatch
+export type Dispatch<State = any> = store.Dispatch<State>
+export type Store<State = any> = store.Store<State>
+
+export type ThunkAction<ReturnType, State = any> = thunks.ThunkAction<ReturnType, State>
+export type BoundThunkDispatch<State = any> = thunks.BoundThunkDispatch<State>
+export type ThunkDispatch<State = any> = thunks.ThunkDispatch<State>
+export type ThunkfulStore<State = any> = thunks.ThunkfulStore<State>
```

Comment:
Some of these types seem a little more specific than is necessary for the top-level interface. How would consumers of Rodux use these types?

For example, it looks like the upstream redux-toolkit only exports ThunkAction and ThunkDispatch from its thunk implementation: https://github.com/reduxjs/redux-toolkit/blob/master/packages/toolkit/src/index.ts#L19

Reply:
Worth noting that any types we export should be treated as part of our API surface, so we should only be exporting types that are explicitly intended for external consumption; otherwise, we end up with parts of our internal type surface having a much stricter contract to uphold

Reply:
BoundThunkDispatch exists for the same reason that I stated below. ThunkfulStore just ended up being a very useful abstraction when needing to annotate the `store` argument by reducing a lot of boilerplate. The types I ended up with here was me trying to balance the size, ergonomics, extensibility, etc of the type surface, but I'm open to reducing the types we export if we feel some are unnecessary.

As an overview:
* ThunkAction is the type of the actual Thunks
* ThunkDispatch/BoundThunkDispatch is used in mapDispatchToProps <-- this one may need to be moved elsewhere
* ThunkfulStore is used to annotate `store` in a thunk

Reply:
The last one is interesting. From what I see in https://github.com/reduxjs/redux-thunk, it seems like if we align to upstream we'd end up with a thunk signature that just gets bound `dispatch` and `getState` functions instead of the whole store, right?

Either way, I do think that the `Bound` stuff belongs in RoactRodux. The rest is probably good, and may change if we ever redefine the thunk signature

Reply:
Correct, ThunkfulStore is only needed because we aren't currently aligned with upstream. I'll remove the bound ones and make a follow up PR for RoactRodux once this one is merged.